core/store/src/trie/shard_tries.rs

```diff
@@ -244,8 +251,13 @@ impl ShardTries {
     ) {
         let mut ops = Vec::with_capacity(deletions.len());
         for TrieRefcountSubtraction { trie_node_or_value_hash, rc, .. } in deletions.iter() {
-            let key = TrieCachingStorage::get_key_from_shard_uid_and_hash(
-                shard_uid,
+            let shard_uid_db_prefix = match self.0.shard_uid_to_db_prefix.get(&shard_uid) {
+                Some(mapped_shard_uid) => *mapped_shard_uid,
+                // TODO(reshardingV3) Think about how None should be handled here.
+                None => shard_uid.into(),
+            };
```

Comment:
This is on a bit higher level than I anticipated. I thought the mapping would happen closer to the db itself, perhaps in Store or Database. I'm not saying this is wrong but I'm curious on your thoughts about how those two approaches compare.